cmd/query/app/server_test.go

```diff
@@ -53,10 +62,314 @@ func TestCreateTLSServerError(t *testing.T) {
+		// 		CertPath:     "invalid/path",
+		// 		KeyPath:      "invalid/path",
+		// 		ClientCAPath: "invalid/path",
+		// 	},
+		// 	clientTLS: tlscfg.Options{
+		// 		Enabled:    true,
+		// 		CertPath:   "invalid/path",
+		// 		KeyPath:    "invalid/path",
+		// 		CAPath:     "invalid/path",
+		// 		ServerName: "example.com",
+		// 	},
+		// 	expectError:      false,
+		// 	expectServerFail: false,
+		// },
+		{
+			name: "Should pass with insecure connection",
+			serverTLS: tlscfg.Options{
+				Enabled: false,
+			},
+			clientTLS: tlscfg.Options{
+				Enabled: false,
+			},
+			expectError:      false,
+			expectServerFail: false,
+		},
+		{
+			name: "should fail with TLS client to untrusted TLS server",
+			serverTLS: tlscfg.Options{
+				Enabled:  true,
+				CertPath: testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:  testCertKeyLocation + "/example-server-key.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				ServerName: "example.com",
+			},
+			expectError:      true,
+			expectServerFail: false,
+		},
+		{
+			name: "should fail with TLS client to trusted TLS server with incorrect hostname",
+			serverTLS: tlscfg.Options{
+				Enabled:  true,
+				CertPath: testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:  testCertKeyLocation + "/example-server-key.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				CAPath:     testCertKeyLocation + "/example-CA-cert.pem",
+				ServerName: "nonEmpty",
+			},
+			expectError:      true,
+			expectServerFail: false,
+		},
+		{
+			name: "should pass with TLS client to trusted TLS server with correct hostname",
+			serverTLS: tlscfg.Options{
+				Enabled:  true,
+				CertPath: testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:  testCertKeyLocation + "/example-server-key.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				CAPath:     testCertKeyLocation + "/example-CA-cert.pem",
+				ServerName: "example.com",
+			},
+			expectError:      false,
+			expectServerFail: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			serverOptions := &QueryOptions{TLSHTTP: test.serverTLS}
+			httpServer, err := createHTTPServer(&querysvc.QueryService{}, serverOptions, opentracing.NoopTracer{}, zap.NewNop())
+
+			if test.expectServerFail {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.NotNil(t, httpServer)
+			}
+			httpListener, err := net.Listen("tcp", "localhost:"+fmt.Sprintf("%d", ports.QueryHTTP))
+			if test.expectServerFail {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+
+			if serverOptions.TLSHTTP.Enabled {
+
+				tlsCfg, _ := serverOptions.TLSHTTP.Config()
+				tlsCfg.Rand = rand.Reader
+				tlsHTTPListener := tls.NewListener(httpListener, tlsCfg)
+				go func() {
+					// err
+					_ = httpServer.Serve(tlsHTTPListener)
+
+				}()
+
+				time.Sleep(10 * time.Millisecond) // wait for server to start serving
```

Comment:
> time.Sleep(10 * time.Millisecond) // wait for server to start serving

sleeps like this usually lead to flaky/unstable tests in CI. If you check other unit tests in this project, the usual approach is to check for condition periodically until it is satisfied or a much larger time period has passed, e.g.
```
for i:=0; i < 100; i++ {
    if conditionSatisfied() {
        break
    }
    sleep 10ms
}
assert condition is satisfied
```

this loop does not increase unit test run time if CPU resources are available, but in CI environment, where CPU is shared and busy, it can wait long enough (1s here) to succeed on subsequent iterations.